### PR TITLE
🔀 Add sentry module

### PR DESCRIPTION
**Pull request by williamchong007**
maybe useful for QA/testing
all config should be configurable by ENV

### CODE REVIEW

**Comment by williamchong007** [comment]
updated

**Comment by rickmak** [comment]
Please add the related ENVVAR to docker-compose as hints to configuration.

**Comment by rickmak** [comment]
I just take a look at https://github.com/nuxt-community/sentry-module the README is only about setup.

How about usage? It injects console or we will use raven directly?

**Comment by williamchong007** [comment]
For env, minimally only `SENTRY_DSN` is needed, added to docker compose file.

It installs raven in client side, and also hooks into nuxt error hook
https://github.com/nuxt-community/sentry-module/blob/master/lib/sentry.js#L55

so everything listed here, including `console.error`/`uncaught exception` should work automagically
https://docs.sentry.io/clients/javascript/#breadcrumbs

In a Vue component, it is also available as `this.$raven`, so we can call functions like 

```javascript
this.$raven.setUserContext({user})
```

where `this` is a Vue instance.

**Comment by williamchong007** [comment]
please dont merge yet, want to confirm if i need to inject this env during webpack build for console.error to work

File: Dockerfile.production [hunk 1]
```diff
@@ -6,6 +6,8 @@ COPY package.json package-lock.json /usr/src/app/
 
 RUN npm ci
 
+ARG SENTRY_DSN
```
Comment (williamchong007) [inline Dockerfile.production]: It seems the `SENTRY_DSN` is needed when building webpack dist, not sure if putting it here is the best way @inDream @rickmak

Comment (rickmak) [reply Dockerfile.production]: Yes, ant put the value into docker-compose.yml

```
build:
  context: .
  args:
    SENTRY_DSN: XXX
```

Comment (williamchong007) [reply Dockerfile.production]: updated and also set the DSN as ENV in Dockerfile, since default server DSN aligns with client build seems more sensible

**Comment by rickmak** [comment]
@inDream want to confirm currently the hub.docker.com auto build is not in use?

I found the image are now build at travis and will override the hub.docker.com auto build.

**Comment by rickmak** [comment]
Sure, may be we need to document a bit how the CI workflow.....

**Comment by rickmak** [comment]
@williamchong007 can you invite me to sentry so I can put the SENTRY_DSN into travis.yml

**Comment by williamchong007** [comment]
@rickmak is it rick.mak@gmail.com ?

**Comment by rickmak** [comment]
Yes

**Comment by williamchong007** [comment]
@rickmak added, plz check if you have enough permission, not rly familiar with sentry org system

**Pull request merged**